egui/src/widgets/text_edit.rs

```diff
@@ -331,6 +334,7 @@ impl<'t> Widget for TextEdit<'t> {
 impl<'t> TextEdit<'t> {
     fn content_ui(self, ui: &mut Ui) -> Response {
         let TextEdit {
+            mut prev_text,
```

Comment:
If we declare `prev_text` here as `let mut prev_text: Option<String> = None` then (`None`)  will mean "no change". We can then later check

``` rust
if let Some(prev_text) = prev_text {
    let info = WidgetInfo::text_changed(prev_text, text);
    self.ctx.output().events.push(OutputEvent::TextChanged(info));
}
````

Reply:
I just made it a clone of `text` so we could do a straight equality check, and also pass it directly to the various text manipulation functions. Does that work for you too? Otherwise agreed that it should be removed from the widget--that's done.

Reply:
I'd rather avoid cloning the text each frame, but if the alternative is very complicated it's fine (it can be optimized later)